ratis-server/src/main/java/org/apache/ratis/server/impl/DisabledDataStreamFactory.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ratis.server.impl;
+
+import org.apache.ratis.client.DataStreamClientFactory;
+import org.apache.ratis.client.DataStreamClientRpc;
+import org.apache.ratis.conf.Parameters;
+import org.apache.ratis.conf.RaftProperties;
+import org.apache.ratis.datastream.SupportedDataStreamType;
+import org.apache.ratis.protocol.RaftPeer;
+import org.apache.ratis.server.DataStreamServerFactory;
+import org.apache.ratis.server.DataStreamServerRpc;
+import org.apache.ratis.server.RaftServer;
+import org.apache.ratis.statemachine.StateMachine;
+
+import java.io.IOException;
+import java.util.Collection;
+
+/** A stream factory that does nothing when data stream is disabled. */
+public class DisabledDataStreamFactory implements DataStreamServerFactory, DataStreamClientFactory {
+    public DisabledDataStreamFactory(Parameters parameters){}
+
+    @Override
+    public DataStreamClientRpc newDataStreamClientRpc(RaftPeer server, RaftProperties properties) {
```

Comment:
These few lines and some other lines below still have 4-space indentation.

Reply:
Oops, this comment is incorrect.  Please ignore it.

Reply:
You are right that I was confused by the GitHub ui.  The comment "These few lines and some other lines below still have 4-space indentation."  is correct.  Please use 2-space indentation.

Reply:
I see what you mean now. Done.

Reply:
Oops... I must have forgot to push my last local commit that fixes indents before going to sleep. Sorry :)